transformer_engine/pytorch/csrc/comm_gemm_overlap.h

```diff
@@ -0,0 +1,579 @@
+/*************************************************************************
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <cuda.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <torch/cuda.h>
+#include <torch/custom_class.h>
+#include <torch/extension.h>
+#include <torch/types.h>
+#include <transformer_engine/userbuffers.h>
+
+#define HALF_BYTES 2
+
+#define CHECK_CUDA(call)                                                                     \
+  do {                                                                                       \
+    cudaError_t status_ = call;                                                              \
+    if (status_ != cudaSuccess) {                                                            \
+      fprintf(stderr, "CUDA Error at line %d: %s\n", __LINE__, cudaGetErrorString(status_)); \
+      exit(1);                                                                               \
+    }                                                                                        \
+  } while (0)
+
+namespace ubuf {
+
+enum COMM_TYPE { RS = 0, AG = 1 };
+
+enum UBOverlapAlgo {
+  BULK_OVERLAP_AG = 0,
+  BULK_OVERLAP_RS = 1,
+  SPLIT_PIPELINED_AG = 2,
+  SPLIT_PIPELINED_RS = 3
+};
```

Comment:
I see we are accessing these enums as enum classes rather than as plain enums:
```suggestion
enum class COMM_TYPE { RS = 0, AG = 1 };

enum class UBOverlapAlgo {
  BULK_OVERLAP_AG = 0,
  BULK_OVERLAP_RS = 1,
  SPLIT_PIPELINED_AG = 2,
  SPLIT_PIPELINED_RS = 3
};
```